vsock-bridge/src/socket.cpp

```diff
@@ -17,142 +17,86 @@ namespace vsockio
+        bool canSendModeData = false;
+		if (!_outputClosed) {
+            if (!_buffer.consumed()) {
+                canSendModeData = send(_buffer);
+                if (_buffer.consumed())
+                {
+                    _buffer.reset();
+                }
+            }
+        }
+
+		if (_peer->closed() && _buffer.consumed())
 		{
-			std::unique_ptr<Buffer>& buffer = _sendQueue.front();
-
-			// received termination signal from peer
-			if (buffer->empty())
-			{
-				Logger::instance->Log(Logger::DEBUG, "[socket] writeToOutput dequeued a termination buffer (fd=", _fd, ")");
-				_sendQueue.dequeue();
-				close();
-				break;
-			}
-			else
-			{
-				send(*buffer);
-				if (buffer->consumed())
-				{
-					_sendQueue.dequeue();
-					_queueFull = false;
-				}
-			}
+            Logger::instance->Log(Logger::DEBUG, "[socket] writeToOutput finished draining socket, closing (fd=", _fd, ")");
+            close();
 		}
 
-		if (_peer->closed())
-		{
-			if (_sendQueue.empty())
-			{
-				Logger::instance->Log(Logger::DEBUG, "[socket] writeToOutput detected input peer is closed, closing (fd=", _fd, ")");
-				close();
-			}
-			else if (!_peer->queueEmpty())
-			{
-				// Peer has some queued data they never received
-				// Assuming this data is critical for the protocol, it should be ok to abort the connection straight away
-				Logger::instance->Log(Logger::DEBUG, "[socket] writeToOutput detected input peer is closed while having data remaining, closing (fd=", _fd, ")");
-				close();
-			}
-		}
-
-		return _sendQueue.empty();
+		return canSendModeData;
 	}
 
-	void Socket::queue(std::unique_ptr<Buffer>&& buffer)
+	bool Socket::read(Buffer& buffer)
 	{
-		_sendQueue.enqueue(std::move(buffer));
-
-		// to simplify logic we allow only 1 buffer for socket sinks
-		_queueFull = true;
-	}
-
-	std::unique_ptr<Buffer> Socket::read()
-	{
-		std::unique_ptr<Buffer> buffer{ BufferManager::getBuffer() };
-
-		while (true)
-		{
-			const int bytesRead = _impl.read(_fd, buffer->tail(), buffer->remainingCapacity());
-			int err = 0;
-			if (bytesRead > 0)
-			{
-				// New content read
-				// update byte count and enlarge buffer if needed
-
-				//Logger::instance->Log(Logger::DEBUG, "[socket] read returns ", bytesRead, " (fd=", _fd, ")");
-				buffer->produce(bytesRead);
-				if (!buffer->ensureCapacity())
-				{
-					break;
-				}
-			}
-			else if (bytesRead == 0)
-			{
-				// Source closed
-
-				Logger::instance->Log(Logger::DEBUG, "[socket] read returns 0, closing input (fd=", _fd, ")");
-				closeInput();
-				break;
-			}
-			else if ((err = errno) == EAGAIN || err == EWOULDBLOCK)
-			{
-				// No new data
-
-				_inputReady = false;
-				break;
-			}
-			else
-			{
-				// Error
-
-				Logger::instance->Log(Logger::WARNING, "[socket] error on read, closing input (fd=", _fd, "): ", strerror(err));
-				closeInput();
-				break;
-			}
-		}
-
-		return buffer;
+        if (!buffer.hasRemainingCapacity()) return false;
```

Comment:
indentation?

Reply:
Fixed tabs to spaces in whole file. Should add clang-style in a future PR.